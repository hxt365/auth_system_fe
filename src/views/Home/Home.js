import React, { useContext } from 'react';
import { Result, Icon } from 'antd';
import './Home.scss';
import { AppContext } from 'components/AppLayout';

function Home(props) {
  const state = useContext(AppContext);
  const { firstName, lastName } = state ? state.user : props;

  return (
    <section className="home">
      <Result
        icon={<Icon type="smile" theme="twoTone" className="home__result" />}
        title={`Great, ${`${firstName} ${lastName}`}! 
        We are here together.`}
      />
    </section>
  );
}

export default Home;
