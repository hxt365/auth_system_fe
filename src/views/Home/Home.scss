.home {
  &__result {
    padding-top: 15rem;
  }
}
