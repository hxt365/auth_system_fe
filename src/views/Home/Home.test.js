import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Home from './Home';

const props = {
  firstName: 'abc',
  lastName: 'xyz',
};

describe('test Home', () => {
  it('test snapshot', () => {
    const wrapper = shallow(<Home {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
