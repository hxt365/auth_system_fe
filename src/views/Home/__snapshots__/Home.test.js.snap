// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Home test snapshot 1`] = `
<section
  className="home"
>
  <OriginResult
    icon={
      <Icon
        className="home__result"
        theme="twoTone"
        type="smile"
      />
    }
    status="info"
    title="Great, abc xyz! 
        We are here together."
  />
</section>
`;
